### ISSUE TITLE
@lerna/package-graph does not work with Lerna v7

### ISSUE THREAD

**Issue by frbuceta** [lerna/lerna#3721]
## Current Behavior

The new configuration is not compatible with @lerna/package-graph
![image](https://github.com/lerna/lerna/assets/13822438/975e0896-fb3c-42f2-86e4-039ea47ba67a)

## Expected Behavior

Being able to get all the packages from the workspace

## Steps to Reproduce

See https://github.com/loopbackio/loopback-next/blob/master/packages/monorepo/lib/update-ts-project-refs.js.  This package has not been updated to work with Lerna v7.

### Failure Logs / Configuration

<details><summary>lerna.json</summary><p>

```json
{
  "$schema": "node_modules/lerna/schemas/lerna-schema.json",
  "command": {
    "bootstrap": {
      "concurrency": 4,
      "forceLocal": true
    },
    "version": {
      "forcePublish": ["@loopback/cli", "@loopback/docs"],
      "message": "chore: publish release",
      "conventionalCommits": true
    },
    "run": {
      "loglevel": "warn",
      "concurrency": 8,
      "prefix": false,
      "stream": true,
      "sort": false
    },
    "clean": {
      "loglevel": "warn",
      "concurrency": 8,
      "yes": true
    }
  },
  "version": "independent"
}
```

</p></details>

### Environment

```
System:
  OS: Linux 6.1 Manjaro Linux
  CPU: (8) x64 Intel(R) Core(TM) i7-4770K CPU @ 3.50GHz
Binaries:
  Node: 20.2.0 - /usr/bin/node
  Yarn: 1.22.19 - /usr/bin/yarn
  npm: 9.6.7 - /usr/bin/npm
Utilities:
  Git: 2.40.1 - /usr/bin/git
npmPackages:
  lerna: ^7.0.0 => 7.0.0 
```

**Comment by JamesHenry** [comment]
Hi @frbuceta, that's correct, the literally dozens of extraneous packages that lerna used to publish are no longer supported.

You can technically already use `@nx/devkit` today to reimplement your logic, but I think this case will be common enough that we'll expose a public API from the lerna package for this.

I'll update this issue once it has been released!

**Comment by JamesHenry** [comment]
@frbuceta `detectProjects` from `lerna/utils` is now available in lerna v7.1.0

You can see the PR that implemented it here: https://github.com/lerna/lerna/pull/3740
